keps/sig-architecture/3935-oldest-node-newest-control-plane/README.md

```diff
@@ -0,0 +1,796 @@
+it became clear that overlap between the oldest and newest version is required to actually give users a version they can use for a year.
+This led to the current policy of supporting each minor version for approximately 14 months, giving users a 2-month period to qualify and upgrade following a new Kubernetes release
+before the oldest supported minor version reaches end-of-life. During that 2-month period, there are actually 4 supported minor versions.
+
+Updating the skew policy to allow up to n-3 nodes brings the policy back in line with its original goals.
+
+### Goals
+
+* Allow users to make full use of the [yearly support period](https://github.com/kubernetes/enhancements/tree/master/keps/sig-release/1498-kubernetes-yearly-support-period)
+* Streamline upgrade paths for nodes / node pools to minimize required workload churn
+* Validate oldest supported node components work with latest control plane components
+* Update the [Kubernetes version skew policy](https://kubernetes.io/releases/version-skew-policy/) to indicate node components from the oldest supported minor version work with control plane components from the newest supported minor version, starting with Kubernetes `v1.25` nodes and `v1.28` control planes.
+
+### Non-Goals
+
+* This does not propose any changes to the number of supported minor versions
+* This does not propose any changes to the minor version support duration
+* This does not propose any changes to the Kubernetes release process
+* This does not propose any changes to the mechanics control plane components use to enable features 
+  and work with older nodes without features ([KEP-3920](https://github.com/kubernetes/enhancements/pull/3920) could help with this)
+* This does not propose delaying introduction of existing features SIGs have planned around existing n-2 node/control plane skew
+* This does not address increasing node skew supported by cluster management tools like `kubeadm`. Maintainers of those tools
+  can choose to make use of the additional supported skew if they wish, but are not required to. For example, `kubeadm` currently officially
+  supports managing [kubelets up to one version older](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/#version-skew-policy)
+  than the `kubeadm` instance, and does not make use of the existing supported n-2 skew in upgrade plans.
+* This does not address third-party components that version themselves with node components and limit skew relative to the control plane.
+  Maintainers of those components can choose to support larger or smaller version skew. Node operators should limit themselves to skew
+  supported by core Kubernetes node components and any third-party components required for node operation.
+
+## Proposal
+
+1. Add a configuration to https://testgrid.k8s.io/sig-node-kubelet#Summary to begin testing 3 version node / control plane skew
```

Comment:
@liggitt https://testgrid.k8s.io/sig-node-kubelet#kubeadm-kinder-kubelet-1-26-on-1-27 is for `run kubeadm-e2e and the conformance suite` and not the full `NodeConformance` (compare with https://testgrid.k8s.io/sig-node-release-blocking#ci-node-e2e). I don't know if just conformance is enough ... leaving this note for @derekwaynecarr @SergeyKanzhelev et al to confirm what we have is enough or not.

Reply:
we have a big gap in previous versions testing. Only making smoke tests. I'm not sure we want to block this KEP by lack of testing. This KEP is not making things worse. The only thing coming to mind in terms of testing gap related to this KEP is maybe a way to run skew tests with api fields enabled/disabled.

Reply:
> This KEP is not making things worse.

That's my perspective as well. I'm happy to collect a list of things SIGs feel they need to improve test coverage for (sig-network mentioned testing kube-proxy skew relative to control plane _and_ relative to kubelet), though I'm not sure if that belongs in this KEP or in an umbrella test-improvement issue

> The only thing coming to mind in terms of testing gap related to this KEP is maybe a way to run skew tests with api fields enabled/disabled.

Interesting... new API fields aren't actually something I would expect older kubelets to have issues with ... old kubelets won't be aware of them or try to populate them, kubelets make patch requests as much as possible rather than straight update requests to minimize impact of unrelated API fields, and the server handling requests from old clients unaware of new fields are actually pretty easy to do as unit/integration tests independent of kubelets.

Reply:
👍🏾